src/ServiceManager/Travely.ServiceManager.UnitTests/ActivityManagerTests.cs

```diff
@@ -0,0 +1,253 @@
+                .Returns(Task.FromResult((ActivityDbModel)null));
+
+            var activity = new Activity(Mocks.Activity)
+            {
+                Id = null
+            };
+
+            activity.Type.Id = null;
+            // Act, Assert
+            await _target.CreateActivityAsync(Mocks.Activity);
+            _unitOfWorkMock.Verify(uow => uow.SaveAsync(), Times.Once);
+        }
+
+        [Fact]
+        public async void CreateActivity_NewActivityWithExistingType_ShouldNotThrow()
+        {
+            // Arrange
+            _activityTypeRepoMock.Setup(repo => repo.GetActivityTypeAsync(It.IsAny<long>(), It.IsAny<string>()))
+                .ReturnsAsync((long agencyId, string typeName) =>
+                {
+                    var activityType = new ActivityType(Mocks.ActivityType)
+                    {
+                        AgencyId = agencyId,
+                        ActivityName = typeName
+                    };
+
+                    return _mapper.Map<ActivityTypeDbModel>(activityType);
+                });
+
+            _activityRepoMock.Setup(repo => repo.GetActivityAsync(It.IsAny<long>(), It.IsAny<string>(), It.IsAny<long>()))
+                .Returns(Task.FromResult((ActivityDbModel)null));
+
+            var activity = new Activity(Mocks.Activity)
+            {
+                Id = null
+            };
+
+            // Act, Assert
+            await _target.CreateActivityAsync(Mocks.Activity);
+            _unitOfWorkMock.Verify(uow => uow.SaveAsync(), Times.Once);
+        }
+
+        [Fact]
+        public async void CreateActivity_ExistingActivity_ShouldThrow()
+        {
+            // Arrange
+            _activityTypeRepoMock.Setup(repo => repo.GetActivityTypeAsync(It.IsAny<long>(), It.IsAny<string>()))
+                .ReturnsAsync((long agencyId, string typeName) =>
+                {
+                    var activityType = new ActivityType(Mocks.ActivityType)
+                    {
+                        AgencyId = agencyId,
+                        ActivityName = typeName
+                    };
+
+                    return _mapper.Map<ActivityTypeDbModel>(activityType);
+                });
+
+            _activityRepoMock.Setup(repo => repo.GetActivityAsync(It.IsAny<long>(), It.IsAny<string>(), It.IsAny<long>()))
+                .ReturnsAsync((long agencyId, string typeName, long activityTypeId) =>
+                {
+                    var activity = new Activity(Mocks.Activity);
+                    activity.Type.AgencyId = agencyId;
+                    activity.Type.ActivityName = typeName;
+                    activity.Type.Id = activityTypeId;
+
+                    return _mapper.Map<ActivityDbModel>(activity);
+                });
+
+            var activity = new Activity(Mocks.Activity);
+
+            // Act, Assert
+            await Assert.ThrowsAnyAsync<InvalidOperationException>(() => _target.CreateActivityAsync(Mocks.Activity));
```

Comment:
You can separate Act & Assert
```suggestion
            Task act() => _target.CreateActivityAsync(Mocks.Activity);

            // Assert
            await Assert.ThrowsAnyAsync<InvalidOperationException>(act);
```

Reply:
The same for the other parts.

Reply:
Done